python_modules/dagster/dagster/_cli/project.py

```diff
@@ -0,0 +1,80 @@
+import os
+
+import click
+
+from dagster._annotations import experimental
+from dagster._generate import generate_project, generate_repository
+
+
+@click.group(name="project")
+def project_cli():
+    """
+    Commands for scaffolding new Dagster projects.
+    """
+
+
+@project_cli.command(
+    name="scaffold-repository",
+    help="Create a new Dagster repository using the default template",
+)
+@click.option(
+    "--name",
+    required=True,
+    type=click.STRING,
+    help="Name of the new Dagster repository",
+)
+@experimental
+def scaffold_repository_command(name: str):
+    """
+    Create a new Dagster project using the default template.
+    """
+    dir_abspath = os.path.abspath(name)
+    if os.path.isdir(dir_abspath) and os.path.exists(dir_abspath):
+        raise FileExistsError(
+            f"""
+            The directory {dir_abspath} already exists. Please delete the contents of this path
+            or choose another repository location.
+            """
+        )
+
+    generate_repository(name)
+    click.echo(_styled_success_statement(name, dir_abspath))
+
+
+@project_cli.command(
+    name="scaffold",
+    help="Create a new Dagster project using the default template",
```

Comment:
Same as above, I think we should be crystal clear that this is creating files and folders. Otherwise people might think it's some sort of action that mutates the Dagster instance or something.

Reply:
<img width="1012" alt="image" src="https://user-images.githubusercontent.com/4531914/181870827-e9bf20e0-43bd-40c3-9f65-9f569bd151ea.png">